modules/service/src/main/scala/lucuma/odb/service/ItcService.scala

```diff
@@ -343,24 +343,24 @@ object ItcService {
           client
             .imaging(targets.imagingInput, useCache = false)
             .map:
-              _.targetTimes.modifyValue:
-                _.map:
-                  _.modifyValue:
-                    case Left(lucuma.itc.Error.SourceTooBright(_))    =>
-                      TargetIntegrationTime(
-                        Zipper.one(IntegrationTime(min, 1.refined)),
-                        Band.R.asLeft, // Band is meaningless here, but we need to provide one
-                        None // Imaging doesn't return signal-to-noise at
-                      ).asRight
-                    case Right(r) if r.times.focus.exposureTime > max =>
-                      r.copy(times = r.times.map(_.copy(exposureTime = max))).asRight
-                    case Right(r) if r.times.focus.exposureTime < min =>
-                      r.copy(times = r.times.map(_.copy(exposureTime = min))).asRight
-                    case other => other
-              .partitionErrors
-              .leftMap(convertErrors(targets))
-
-        targets.imaging.mode match
+              _.all.head.targetTimes.modifyValue:
+                  _.map:
+                    _.modifyValue:
+                      case Left(lucuma.itc.Error.SourceTooBright(_))    =>
+                        TargetIntegrationTime(
+                          Zipper.one(IntegrationTime(min, 1.refined)),
+                          Band.R.asLeft, // Band is meaningless here, but we need to provide one
+                          None // Imaging doesn't return signal-to-noise at
+                        ).asRight
+                      case Right(r) if r.times.focus.exposureTime > max =>
+                        r.copy(times = r.times.map(_.copy(exposureTime = max))).asRight
+                      case Right(r) if r.times.focus.exposureTime < min =>
+                        r.copy(times = r.times.map(_.copy(exposureTime = min))).asRight
+                      case other => other
+                .partitionErrors
+                .leftMap(convertErrors(targets))
+
+        targets.imaging.modes.head match
```

Comment:
I can't follow this code in the diff. Why are we only using the `head` here? Is there anything that says all of the modes in the list have to be the same?

Reply:
this is the minimum to get it to compile in general, but going beyond science confirmed that we do acquisition with a single filter. I should add an assert as for spectroscopy

Reply:
I'm confused, why is there more than one mode?  Imaging supports multiple filters but it is one mode right?

Reply:
right, so it comes down to two aspects:
* old itc doesn't have the concept of multifilters, it will do n calculations with n results, we could model that as imaging modes with n filters or n imaging modes, internally it will have to do n calculations anyway
* from explore is more handy to pass multiple modes each with one filter, as we display a row per filter.

 I think it is useful in general to support requesting n modes but I could be convinced otherwise